include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8874,6 +8874,15 @@ def err_bounds_type_annotation_lost_checking : Error<
   def err_expected_bounds : Error<
     "expression has no bounds">;
 
+  def err_expected_bounds_for_ptr_cast : Error<
+    "expression has no bounds, cast to ptr<T> require a source /w bounds">; 
+
+  def err_expected_bounds_for_assignment : Error<
+    "expression has no bounds, lvalue /w bounds require a rvalue /w bounds">;
```

Comment:
I worry that some C programmers won't understand this error message because they have an imprecise grasp of lvalue vs. rvalue terminology.  Is there to reword this?   For example, "right-hand side of assignment expected to have bounds because the left-hand side lvalue has bounds."